### PR TITLE
Feature/holo 530 holograph genesis tests

### PR DESCRIPTION
**Pull request by alexanderattar**
## Describe Changes

- Add tests for HolographGenesis
- Added MockHolographGenesis contract for testing inherited functions
- Added empty Mock contract for testing external calls
- Prettier `yarn prettier:fix`

image

image

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Hardhat tests are passing

### CODE REVIEW

File: src/mock/MockHolographGenesisChild.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import "../HolographGenesis.sol";
+
+contract MockHolographGenesisChild is HolographGenesis {
+  constructor() {}
+
+  function approveDeployerMock(address newDeployer, bool approve) external onlyDeployer {
+    return this.approveDeployer(newDeployer, approve);
+  }
+
+  function IsApprovedDeployerMock(address deployer) external view returns (bool) {
```
Comment (alexanderattar) [inline src/mock/MockHolographGenesisChild.sol]: lowercase the Is

File: deploy/02_sources.ts [hunk 2]
```diff
@@ -530,14 +530,8 @@ const func: DeployFunction = async function (hre1: HardhatRuntimeEnvironment) {
...
           generateInitCode(
-            ['address', 'address', 'address', 'address', 'address'],
-            [
-              futureBridgeProxyAddress,
-              futureHolographAddress,
-              futureHolographInterfacesAddress,
-              futureRegistryProxyAddress,
-              futureHlgAddress,
-            ]
+            ['address', 'address', 'address', 'address'],
```
Comment (ACC01ADE) [inline deploy/02_sources.ts]: @alexanderattar I think there is older deploy scripts code in your commits. These changes would remove additional data for contracts in deployments and break everything. You might need to check the `experimental` branch and copy from it the contents of the `deploy/02_sources.ts` file.

File: src/mock/MockHolographGenesisChild.sol (hunk 1 above)
Comment (alexanderattar) [reply src/mock/MockHolographGenesisChild.sol]: this is done

File: deploy/02_sources.ts [hunk 3]
```diff
@@ -245,22 +245,17 @@ const func: DeployFunction = async function (hre1: HardhatRuntimeEnvironment) {
     'Holograph Utility Token',
     '1',
     18,
-    ConfigureEvents([HolographERC20Event.bridgeIn, HolographERC20Event.bridgeOut]),
+    ConfigureEvents([]),
     generateInitCode(['address'], [deployer.address]),
     salt
   );
 
   const holographerBytecode: BytesLike = (await hre.ethers.getContractFactory('Holographer')).bytecode;
```
Comment (alexanderattar) [inline deploy/02_sources.ts:253]: @ACC01ADE is this change intended?

**Review by ackvf — CHANGES_REQUESTED** [review]
Please remove stylistic (prettier) changes to unrelevant files you didn't directly touch. This is going to cause a lot of headaches to other pull requests.

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**